theano/tensor/opt.py

```diff
@@ -5157,10 +5162,9 @@ def local_sum_prod_mul_by_scalar(node):
                 new_op_input_nb_elements = new_op_input.size
                 new_op_output = node.op(new_op_input)
 
-            if not len(non_scalars) == 0:
-                # Copy over stacktrace from previous output to new mul op,
-                # for same reason as above.
-                copy_stack_trace(node.outputs, new_op_output)
+            # Copy over stacktrace from previous output to new mul op,
+            # for same reason as above.
+            copy_stack_trace(node.outputs, new_op_output)
```

Comment:
That change may have been introduced by the rebase, but is not correct: if `len(non_scalars)` equals 0, then `new_op_output` is the Python literal `1`, which does not have a tag or stack trace.